From 040ddc03f6660b15d58f3c23613053ebca85e49d Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kai.uwe.broulik@basyskom.com>
Date: Tue, 7 Oct 2025 16:29:02 +0200
Subject: [PATCH 325/553] QPdf: Consider devicePixelRatio in alphalessImage in
 PDF/A-1b

The new image must have the same device pixel ratio as the original,
otherwise it will not be painted correctly.

Pick-to: 6.8
Change-Id: I4d885c0f819e544a6717e4ef5bf8873bd75af23c
Reviewed-by: Lars Knoll <lars@knoll.priv.no>
Reviewed-by: Giuseppe D'Angelo <giuseppe.dangelo@kdab.com>
(cherry picked from commit 36983632f8ec4a700586f279d3b1c1c8779f3e47)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/gui/painting/qpdf.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gui/painting/qpdf.cpp b/src/gui/painting/qpdf.cpp
index 6419ef64435..908051a477c 100644
--- a/src/gui/painting/qpdf.cpp
+++ b/src/gui/painting/qpdf.cpp
@@ -3213,6 +3213,7 @@ int QPdfEnginePrivate::addImage(const QImage &img, bool *bitmap, bool lossless,
             // a format without alpha channel first
 
             QImage alphaLessImage(image.width(), image.height(), QImage::Format_RGB32);
+            alphaLessImage.setDevicePixelRatio(image.devicePixelRatioF());
             alphaLessImage.fill(Qt::white);
 
             QPainter p(&alphaLessImage);
-- 
2.51.2

